src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerProductTable.cpp

```diff
@@ -0,0 +1,92 @@
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+#include <U2Core/AnnotationSelection.h>
+#include <U2Core/AnnotationTableObject.h>
+#include <U2Core/DNASequenceSelection.h>
+#include <U2Core/U2SafePoints.h>
+
+#include <U2View/ADVSequenceObjectContext.h>
+#include <U2View/AnnotatedDNAView.h>
+
+#include "PCRPrimerProductTable.h"
+#include "src/tasks/PCRPrimerDesignForDNAAssemblyTask.h"
+
+namespace U2 {
+
+ PCRPrimerProductTable::PCRPrimerProductTable(QWidget *parent)
+    : QTableWidget(parent) {
+     setColumnCount(2);
+     setRowCount(8);
+     setHorizontalHeaderLabels(QStringList() << tr("Fragment") << tr("Region"));
+     setSelectionBehavior(QAbstractItemView::SelectRows);
+     setSelectionMode(QAbstractItemView::SingleSelection);
+     connect(selectionModel(), SIGNAL(selectionChanged(const QItemSelection &, const QItemSelection &)), SLOT(sl_selectionChanged()));
+     connect(this, SIGNAL(clicked(const QModelIndex &)), SLOT(sl_selectionChanged()));
+ }
+
+void PCRPrimerProductTable::setCurrentProducts(const QList<U2Region> &currentProducts, AnnotatedDNAView *_associatedView) {
+    SAFE_POINT(currentProducts.size() == 8, "Should be 8 results", );
+    int index = 0;
+    int row = 0;
+    foreach (const U2Region &region, currentProducts) {
+        if (region != U2Region()) {
+            setItem(row, 0, new QTableWidgetItem(PCRPrimerDesignForDNAAssemblyTask::FRAGMENT_INDEX_TO_NAME.at(index)));
+            setItem(row, 1, new QTableWidgetItem(tr("%1-%2").arg(QString::number(region.startPos)).arg(QString::number(region.endPos()))));
+            row++;
+        }
+        index++;
+    }
+    associatedView = _associatedView;
+}
+
+void PCRPrimerProductTable::setAnnotationTableObject(AnnotationTableObject *ato) {
+    associatedTableObject = ato;
+}
+
+void PCRPrimerProductTable::sl_selectionChanged() {
+    QModelIndexList selectedIndexesList = selectedIndexes();
+    if (selectedIndexesList.isEmpty()) {
+        return;
+    }
+    //one row = 2 items
+    CHECK(selectedIndexesList.size() == 2, );
+    QTableWidgetItem *selectedItem = item(selectedIndexesList.first().row(), 0);
+    QString selectedFragmentName = selectedItem->text();
+    
+    Annotation *selectedAnnotation = nullptr;
+    for (auto a : associatedTableObject->getAnnotations()) {
```

Comment:
Move `associatedTableObject->getAnnotations()` to a separate variable and add `qAsConst`